main.go

```diff
@@ -72,52 +74,34 @@ func main() {
 			os.Exit(1)
 		}
 	}
+
 	ip, err := getHostIp()
 	if err != nil {
 		fmt.Fprintf(os.Stderr, "Warning: failed to determine host's ip: %s\n", err)
-		ip = ""
 	}
+
 	// Retrieve zone from the metadata server only if not specified in args.
 	zone := *localityZone
```

Comment:
What about backward compatibility for the `--locality-zone` flag? That is, if `--locality-zone` is supplied but not `--gke-location`, we should use that as the location of the cluster.

Reply:
I was trying to discuss this in the updated PR description. 

IMO, 

1.  `--locality-zone` should not be overloaded. This flag is being used to specify the `zone` in all use cases of the generator, expect when used with `--generate-mesh-id-experimental`.
    - `--generate-mesh-id-experimental` is marked experimental we should in the clear to change the dependencies.
    -  Adding b/c for the flag would open us to the vulnerability of users specifying a "region" to `--locality-zone` leading to incorrect bootstrap config generation (for eg: Node information [here](https://github.com/GoogleCloudPlatform/traffic-director-grpc-bootstrap/blob/master/main.go#L262)).

1.  Customers wanting the new UX of determining the location of the cluster from MDS would any how explicitly upgrade the version of the generator. In which case, its easy to modify both `image` and `args` when updating k8s Deployment spec.

Reply:
>  In which case, its easy to modify both image and args when updating k8s Deployment spec.

This may be true, but we should discuss with the customer before going that route.

Reply:
Discussed offline, and there is an action item to bring this up in the next customer meeting. Keeping this unresolved to update with what we hear back from the customer.